generative/networks/nets/diffusion_model_unet.py

```diff
@@ -515,12 +516,17 @@ def __init__(
                 conv_only=True,
             )
         else:
-            assert self.num_channels == self.out_channels
+            if self.num_channels != self.out_channels:
+                raise ValueError("num_channels and out_channels must be equal when use_conv=False")
```

Comment:
This requirement should be stated in the docstring for `use_conv` so that this isn't a surprise.